### PR TITLE
[geometry] Faster CellId::XY and CellId::FromXY.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

File: indexer/cell_id.hpp [hunk 1]
```diff
@@ -72,14 +74,14 @@ class CellIdConverter
 
   static m2::PointD FromCellId(CellId id)
   {
-    pair<uint32_t, uint32_t> const xy = id.XY();
+    std::pair<uint32_t, uint32_t> const xy = id.XY();
```
Comment (tatiana-yan) [inline indexer/cell_id.hpp]: `#include <cstdint>`

File: base/base_tests/bits_test.cpp [hunk 2]
```diff
@@ -63,6 +63,32 @@ UNIT_TEST(PerfectShuffle)
...
 
+UNIT_TEST(BitwiseMerge)
+{
+  TEST_EQUAL(bits::BitwiseMerge(1, 1), 3, ());
+  TEST_EQUAL(bits::BitwiseMerge(3, 1), 7, ());
+  TEST_EQUAL(bits::BitwiseMerge(1, 3), 11, ());
+  TEST_EQUAL(bits::BitwiseMerge(uint32_t{1} << 31, uint32_t{1} << 31), uint64_t{3} << 62, ());
+
+  auto bitwiseMergeSlow = [](uint32_t x, uint32_t y) -> uint64_t {
+    uint64_t result = 0;
```
Comment (tatiana-yan) [inline base/base_tests/bits_test.cpp]: `#include <cstdint>`

Comment (mpimenov) [reply base/base_tests/bits_test.cpp]: done

File: indexer/cell_id.hpp (hunk 1 above)
Comment (mpimenov) [reply indexer/cell_id.hpp]: done

**Pull request merged**